src/core/solution.jl

```diff
@@ -66,33 +67,66 @@ end
 
 ""
 function build_solution(aim::AbstractInfrastructureModel; post_processors=[])
-    sol = build_solution_values(aim.sol)
+    sol = Dict{String, Any}("it" => Dict{String, Any}())
+
+    for it in it_ids(aim)
+        it_str = string(it)
+        sol["it"][it_str] = build_solution_values(aim.sol[:it][it])
+    end
 
     solution_preprocessor(aim, sol)
```

Comment:
I think the best convention for this routine should be to apply the solution post-processors here on solution data that has explicit dicts for "it" and "nw".  Downstream packages should be able to use `apply_to_all` types of functions to easily support this case.

Once the post processors are run, the removal of IT and NW dicts to support the implicit cases can be applied.

Reply:
Agreed. If you look at the remainder of this function, that is exactly what's done.